spanner/batch.go

```diff
@@ -291,10 +320,24 @@ func (t *BatchReadOnlyTransaction) Execute(ctx context.Context, p *Partition) *R
 				RequestOptions: p.rreq.RequestOptions,
 				ResumeToken:    resumeToken,
 			})
+			if err != nil {
+				return client, err
+			}
+			md, errGFE := client.Header()
```

Comment:
Hi Knut, the reason why I did this was because in someplaces I was worries that the error was getting returned at the end of the function, example [here ](https://github.com/googleapis/google-cloud-go/blob/5a2ed6b2cd1c304e0f59daa29959863bff9b5c29/spanner/batch.go#L131). So will have to use a separate error in such places. Hence to seem consistent I used errGFE

Reply:
Yeah, I understand that. But I think that it is better to follow the standard naming of Golang when possible, and only make an exception when you need to. In this case for example, you no longer need to return `err` at the end of the function anymore, as you know that it is `nil` because of the `if` statement that was added above. So the last statement in the function can now be changed to `return client, nil`

Reply:
Actually I looked into it further and I realised if I define err inside the loop even in the case I described I can use it as the scope of that variable will be limited to the if condition loop. So i think it should be okay to do that. Wdyt?

Reply:
I'm not really sure I understand what you mean in this case. But if you mean that you can define `err` only inside the if-block, then that would be good.

Reply:
Yes that's what I mean